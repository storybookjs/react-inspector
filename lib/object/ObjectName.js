'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createStyles = require('../styles/createStyles');

var _createStyles2 = _interopRequireDefault(_createStyles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * A view for object property names.
 *
 * If the property name is enumerable (in Object.keys(object)),
 * the property name will be rendered normally.
 *
 * If the property name is not enumerable (`Object.prototype.propertyIsEnumerable()`),
 * the property name will be dimmed to show the difference.
 */
var ObjectName = function ObjectName(_ref, _ref2) {
  var name = _ref.name,
      dimmed = _ref.dimmed;
  var theme = _ref2.theme;

  var styles = (0, _createStyles2.default)('ObjectName', theme);
  return _react2.default.createElement(
    'span',
    { style: (0, _extends3.default)({}, styles.base, dimmed && styles.dimmed) },
    name
  );
};

ObjectName.propTypes = {
  /** Property name */
  name: _react.PropTypes.string,
  /** Should property name be dimmed */
  dimmed: _react.PropTypes.bool
};

ObjectName.defaultProps = {
  dimmed: false
};

ObjectName.contextTypes = {
  theme: _react2.default.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.object])
};

exports.default = ObjectName;