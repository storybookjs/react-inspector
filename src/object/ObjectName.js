import React, { Component } from 'react';

// Styles
import objectStyles from './objectStyles';

const ObjectName = ({ name, query }) => {
    if (query.includes(name)){
        return (<span style={objectStyles.query}> { name} </span>)
    } else {
        return (<span style={objectStyles.name}>{ name }</span>)
    }

}


export default ObjectName
