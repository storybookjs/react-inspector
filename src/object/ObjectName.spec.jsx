import React from 'react';
import { render } from '@testing-library/react';
import { ObjectName } from './ObjectName';

import { describe, it, expect } from 'vitest';

describe('ObjectName', () => {
  it('should render', () => {
    const { container } = render(<ObjectName name="testvalue" />);
    expect(container).toMatchSnapshot();
  });

  it('should apply dimming if `dimmed` prop is true', () => {
    const { container } = render(<ObjectName name="testvalue" dimmed={true} />);
    expect(container).toMatchSnapshot();
  });

  it('should not apply dimming if `dimmed` prop is false', () => {
    const { container } = render(<ObjectName name="testvalue" dimmed={false} />);
    expect(container).toMatchSnapshot();
  });

  it('Accepts and applies additional `style` prop', () => {
    // Test that a custom `style` props is passed and applied to <span/>
    const style = { color: 'hotpink' };
    const { container } = render(<ObjectName styles={style} />);
    expect(container).toMatchSnapshot();
  });
});
