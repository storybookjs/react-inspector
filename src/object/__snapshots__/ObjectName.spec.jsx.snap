// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ObjectName > Accepts and applies additional \`style\` prop 1`] = `
<div>
  <span
    style="color: hotpink;"
  />
</div>
`;

exports[`ObjectName > should apply dimming if \`dimmed\` prop is true 1`] = `
<div>
  <span
    style="color: rgb(136, 19, 145); opacity: 0.6;"
  >
    testvalue
  </span>
</div>
`;

exports[`ObjectName > should not apply dimming if \`dimmed\` prop is false 1`] = `
<div>
  <span
    style="color: rgb(136, 19, 145);"
  >
    testvalue
  </span>
</div>
`;

exports[`ObjectName > should render 1`] = `
<div>
  <span
    style="color: rgb(136, 19, 145);"
  >
    testvalue
  </span>
</div>
`;
